pulsar-client-cpp/include/pulsar/Message.h

```diff
@@ -44,14 +44,15 @@ class Message {
     typedef std::map<std::string, std::string> StringMap;
 
     Message();
+    virtual ~Message() {}
 
     /**
      * Return the properties attached to the message.
      * Properties are application defined key/value pairs that will be attached to the message
      *
      * @return an unmodifiable view of the properties map
      */
-    const StringMap& getProperties() const;
+    virtual const StringMap& getProperties() const;
```

Comment:
Why is this change needed? We shouldn't have more types of messages at this point. 

Also, I think this is not binary compatible with current version and would have to be deferred to major version change.

Reply:
@merlimat We need to set these method to virtual because of the restriction of Google Mock.
https://github.com/google/googletest/blob/master/googlemock/docs/ForDummies.md#how-to-define-it